KalturaWowzaServer/src/main/java/com/kaltura/media/server/wowza/CuePointsManager.java

```diff
@@ -276,139 +273,151 @@ public void onLiveStreamPacketizerInit(ILiveStreamPacketizer liveStreamPacketize
-			if(!streams.containsKey(liveEntry.id))
-				throw new KalturaManagerException("Entry media stream not found");
-
-			stream = streams.get(liveEntry.id).get(0);
+	private void startSyncPoints(Timer t, final IMediaStream stream, final String entryId) {
+		TimerTask tt = new TimerTask() {
+			@Override
+			public void run() {
+				try {
+					createSyncPoint(stream,entryId);
+				} catch (Exception e) {
+					logger.error("Error occured while running sync points timer", e);
+				}
+			}
+		};
+		try {
+			t.schedule(tt,0, syncPointsInterval);
+		} catch (Exception e) {
+			logger.error("Error occurred while scheduling a timer task",e);
 		}
+	}
 
-		logger.debug("Live entry duration [" + liveEntry.duration + "] stream elapsed time [" + stream.getElapsedTime().getTimeSeconds() + "]");
-		return (double) (liveEntry.duration + stream.getElapsedTime().getTimeSeconds());
+	private void createSyncPoint(IMediaStream stream, String entryId) {
+		logger.debug("createSyncPoint. entryId:"+entryId);
+		KalturaLiveEntry liveEntry = liveManager.get(entryId);
+		String id = StringUtils.getUniqueId();
+		double offset = liveEntry.duration + stream.getElapsedTime().getTimeSeconds();
+		sendSyncPoint(stream,entryId, id, offset);
 	}
 
-	@Override
-	public void sendSyncPoint(String entryId, String id, double offset) throws KalturaManagerException {
+	public void sendSyncPoint(final IMediaStream stream, String entryId, String id, double offset) {
 		logger.debug("sendSyncPoint: entryId: " + entryId + " id: " + id + " offest: " + offset);
-		final ArrayList<IMediaStream> entryStreams;
-		synchronized (streams) {
-			if(!streams.containsKey(entryId))
-				throw new KalturaManagerException("Entry [" + entryId + "] media stream not found");
-
-			entryStreams = streams.get(entryId);
-		}
 
 		Date date = new Date();
+		long time = date.getTime();
 		AMFDataObj data = new AMFDataObj();
 		data.put("objectType", "KalturaSyncPoint");
 		data.put("id", id);
 		data.put("offset", offset * 1000);
-		data.put("timestamp", date.getTime());
-
-		logger.debug("About to processDirectMessages on: " + entryStreams.size() + " streams");
-		for (IMediaStream stream : entryStreams) {
-			stream.sendDirect(CuePointsManager.PUBLIC_METADATA, data);
-			((MediaStream)stream).processSendDirectMessages();
-			logger.info("Sent sync-point [" + id + "] to entry [" + entryId + "] stream [" + stream.getName() + "] offset [" + offset + "]");
-		}
-
-
-		TimerTask task = new TimerTask() {
-
-			@Override
-			public void run() {
-				AMFDataObj clear = new AMFDataObj();
-				clear.put("objectType", "");
-
-				for (IMediaStream stream : entryStreams) {
-					stream.sendDirect(CuePointsManager.PUBLIC_METADATA, clear);
-					((MediaStream)stream).processSendDirectMessages();
-//					logger.debug("clearing sync point. stream name:" + stream.getName());
-				}
-			}
-		};
-
-		Timer timer = new Timer("sendSyncPoint-" + entryId, true);
-		timer.schedule(task, 1000);
+		data.put("timestamp", time);
```

Comment:
We assume that all machines in all data-centers (including our SaaS PA and NY) are configured to use the same timezone and synchronized between them.